cwltool/process.py

```diff
@@ -453,6 +454,9 @@ def __init__(self, toolpath_object, **kwargs):
                              self.tool.get("requirements", []) +
                              get_overrides(kwargs.get("overrides", []), self.tool["id"]).get("requirements", []))
         self.hints = kwargs.get("hints", []) + self.tool.get("hints", [])
+        # Versions of requirements and hints which aren't mutated.
+        self.original_requirements = copy.deepcopy(self.requirements)
+        self.original_hints = copy.deepcopy(self.hints)
```

Comment:
Wouldn't it be better to use `copy.deepcopy` when `self.requirements` and `self.hints` are originally defined on lines 453 and 456?

Reply:
I take this back, see https://github.com/common-workflow-language/cwltool/issues/740#issuecomment-387966361